src/main/java/Duke.java

```diff
@@ -1,10 +1,101 @@
-public class Duke {
-    public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+import java.io.FileNotFoundException;
+
+import util.*;
+
+/**
+ * Duke's main class.
+ *
+ * Consists of a Storage, TaskList and Ui.
+ * Contains the program loop in run() method.
+ */
+public class Duke{
+    // Duke's program variables
+    private Storage storage;
+    private TaskList tasks;
+    private Ui ui;
+
+    /**
+     * Duke's Default Constructor.
+     * Initializes ui, storage and tasks.
+     * Loads in save file (if any).
+     * Uses a default filePath
+     */
+    public Duke() {
+        ui = new Ui();
+        storage = new Storage("data/tasks.txt");
+        try {
+            tasks = new TaskList(storage.loadFileContents());
+            ui.setGreetings("... Oh! You're back!\nEr, gimme a sec...");
+        } catch (FileNotFoundException e) {
+            ui.setGreetings("... Who? Never mind. Er-hmm.\n");
+            tasks = new TaskList();
+        }
+    }
+
+    /**
+     * Own function to generate a response to user input.
+     * Code based off run().
+     *
+     * @param input User input.
+     */
+    public String getResponse(String input) {
+        // Read user input
+        String userInput = input;
+        String output = "";
+        try {
+            // Parse out the command from the user input
+            Command cmd = Parser.parse(userInput);
+            // Use switch to process the commands
+            switch (cmd) {
+            case EXIT:
+                String error = "\n";
+                // Try to Save the data
+                try {
+                    storage.saveToFile(tasks.getTasks());
+                } catch (DukeException e) {
+                    error = ui.getError(e);
+                } finally {
+                    // Return farewells
+                   return ui.getFarewells() + error;
+                }
```

Comment:
Naming a new line as `error` and then overriding it with an actual error may be confusing. Perhaps consider making separate variables for each.